### PR TITLE
fix: Added date to headers for item sync

### PR DESCRIPTION
**Pull request by ghost**
**Description**
This PR is made to send the date when the asset was taken, to the stack by using headers, like the iOS version.
**How does it work ?**
The date taken from the MediaStore (like PHAsset in iOS) and sent to the JS, the JS sent it back to the method doInBackground to upload the assets, from here the application uses the date in ISO 8601 and converts it in RFC 1123 (HTTP format needed to send the date through headers).
The formatted date is added in the headers and sent when upload has completed.

### CODE REVIEW

**Comment by Crash--** [comment]
Can you do something about the mess in the diff? It's pretty hard to read :/. I think that tab / space are the root cause of that. 

Also, we `parseIso8601DateTime` for instance has nothing to do with the class `ListItem` we should not declare it in that place

File: src/android/io/cozy/plugins/listlibraryitems/ListLibraryItems.java [hunk 1]
```diff
@@ -271,6 +284,36 @@ private static String getMimeType(String url) {
...
+    }
+
+    // This method format a Date to a RFC 1123 String
+    public static String formatRfc1123DateTime(Date date, TimeZone timeZone) {
+        SimpleDateFormat dateFormat = new SimpleDateFormat(RFC_1123_DATE_TIME);
+        if (timeZone != null) {
+            dateFormat.setTimeZone(timeZone);
+        }
+        return dateFormat.format(date);
+    }
```
Comment (Crash--) [inline src/android/io/cozy/plugins/listlibraryitems/ListLibraryItems.java]: can we declare the methods relative to DateManipulation outside of the ListItem class?

Comment (ghost) [reply src/android/io/cozy/plugins/listlibraryitems/ListLibraryItems.java]: Sure, but is it necessary to create a new file, or leave everything on this one ?

Comment (Crash--) [reply src/android/io/cozy/plugins/listlibraryitems/ListLibraryItems.java]: Having it in the same file is enough for now :).

**Comment by nono** [comment]
Ping @Crash--

**Pull request merged**